sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Json.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+/**
+ * Simplify redundant json related expressions.
+ */
+object OptimizeJsonExprs extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+    case p => p.transformExpressionsUp {
+      case JsonToStructs(_, options1, StructsToJson(options2, child, timeZoneId2), timeZoneId1)
```

Comment:
We have a general rule `RemoveNoopOperators` for operators. I'm wondering if we can make a general rule `RemoveNoopExpression` for expressions.

Reply:
Hmm, we already have `SimplifyExtractValueOps`, which is somehow close to you meant. BTW, I will later work on another PR to add column pruning of  `JsonToStructs` to `OptimizeJsonExprs`. It sounds not exactly matching `RemoveNoopExpression`, seems to me.